### PR TITLE
fix: Include full command line to all setups

### PR DESCRIPTION
**Pull request by diegomscoelho**
Jira: [Card](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-5236)

## Description

Full command-line was not added to header if command-line wasn't used. Besides that, web VEP output included a white-space as a full command-line in header. This PR fixes both problems. Default values wasn't included on output.

## Test

1) Test `--config` setup;
2) Test VEP flags;
3) Web VEP (Sandbox in card).

### CODE REVIEW

File: modules/Bio/EnsEMBL/VEP/Config.pm [hunk 1]
```diff
@@ -487,6 +487,22 @@ sub new {
     $config->{cache} = 1;
   }
 
+  my $config_command;
+
+  my @skip_opts = qw(web_output host port stats_file user warning_file input_data);
+
+  foreach my $flag (sort keys %$config) {
+    my $value = $config->{$flag};
+    next if !defined($value) || $value eq 0 || (ref($value) eq "ARRAY" && @{$value} == 0) || grep { /$flag/ } @skip_opts;
```
Comment (nuno-agostinho) [inline modules/Bio/EnsEMBL/VEP/Config.pm:496]: Is there a parameter where the user may be expected to pass a value of `0` that would be ignored by this `if` statement?

I don't think so, but just want to confirm.

File: modules/Bio/EnsEMBL/VEP/Config.pm [hunk 2]
```diff
@@ -487,6 +487,22 @@ sub new {
...
+
+    $value = join(" --$flag ", @{$value}) if ref($value) eq "ARRAY";
```
Comment (nuno-agostinho) [inline modules/Bio/EnsEMBL/VEP/Config.pm:498]: Do all flags share the same name between the internal flag and VEP argument?

File: modules/Bio/EnsEMBL/VEP/Config.pm [hunk 3]
```diff
@@ -487,6 +487,22 @@ sub new {
...
+    $value =~ s/(\/[\w-]+?)+\//\[PATH\]\//g;
+    $config_command .= $value eq 1? "--$flag "  : "--$flag $value ";
```
Comment (nuno-agostinho) [inline modules/Bio/EnsEMBL/VEP/Config.pm:500]: This makes sense if all the flags with a value of `1` are boolean. Are there any flags that receive 1 as a numeric value?

**Review by nuno-agostinho — COMMENTED** [review]
Hey @diegomscoelho, thanks for your PR!

The PR seems to work fine in your web sandbox, as well as to print the expected arguments when using CLI flags, a INI configuration file and environment variables (e.g., `export VEP_PICK=1`).

Still, I put some comments that I would like to get your insight on. Thanks!

File: t/AnnotationSource_File_GFF.t [hunk 4]
```diff
@@ -476,7 +476,7 @@ SKIP: {
 
   $runner->run;
   open IN, $test_cfg->{user_file}.'.out';
-  my @tmp_lines = <IN>;
+  my @tmp_lines = grep {!/^\#/} <IN>;
```
[the remark this answers is not in the document]
Comment (diegomscoelho) [reply t/AnnotationSource_File_GFF.t:479]: This alteration is no longer necessary, removing it.

File: modules/Bio/EnsEMBL/VEP/Config.pm (hunk 3 above)
Comment (diegomscoelho) [reply modules/Bio/EnsEMBL/VEP/Config.pm:500]: Four have possible numeric inputs:
```
--buffer_size [number]
--port [number]
--db_version [number]
--fork [num_forks]
```

I don't think any of those will have `1` as an input.

File: modules/Bio/EnsEMBL/VEP/Config.pm (hunk 2 above)
Comment (diegomscoelho) [reply modules/Bio/EnsEMBL/VEP/Config.pm:498]: Yes. It should not have any conflict with those names.

File: modules/Bio/EnsEMBL/VEP/Config.pm (hunk 1 above)
Comment (diegomscoelho) [reply modules/Bio/EnsEMBL/VEP/Config.pm]: Yes, two flags:
```sh
--shift_hgvs
--clin_sig_allele
```
Refactored code to check `%DEFAULTS` and remove from full command any flag equals to default usage. What you think?

**Review by nuno-agostinho — APPROVED** [review]
Hey @diegomscoelho, I tested the changes and they work for `--shift_hgvs 0 --clin_sig_allele 0`. Seems good to merge, thanks!

**Comment by nuno-agostinho** [comment]
Merged to `main` and `release-109`.

**Pull request merged**